### PR TITLE
Issue/2122 Accordion

### PR DESCRIPTION
**Pull request by mhaegeleMHP**
### Pull Request Checklist

- [x] I have read and accept the [Contributor License Agreement](https://github.com/porscheofficial/oss-docs/blob/67c5b24838a293ce7a964884e6005eb71f2b8579/CONTRIBUTOR_LICENSE_AGREEMENT.md)

#### References

- Preview: https://designsystem.porsche.com/issue/2122/

#### Scope

Short description of the pull request scope

### CODE REVIEW

File: packages/components/src/components/accordion/accordion-styles.ts [hunk 1]
```diff
@@ -24,41 +27,59 @@ export const getComponentCss = (
...
+          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+          // @ts-ignore
+          fontSize: fontSizeText[s],
+          padding: compact ? '4px 0' : `${s === 'medium' ? '20px' : '12px'} 0`,
+        })),
+        '&::before': {
+          content: '""',
+          position: 'absolute',
+          borderRadius: borderRadiusSmall,
+          transition: getTransition('background-color'),
```
Comment (denyo) [inline packages/components/src/components/accordion/accordion-styles.ts:57]: Only relevant for hover? So why not move it into the hover media query.

File: packages/components/src/components/accordion/accordion-styles.ts [hunk 2]
```diff
@@ -73,9 +94,9 @@ export const getComponentCss = (
     icon: {
       width: fontLineHeight,
       height: fontLineHeight,
-      marginLeft: '1.5rem',
+      marginLeft: '24px',
```
Comment (denyo) [inline packages/components/src/components/accordion/accordion-styles.ts:97]: Could use flex gap?

File: packages/components/CHANGELOG.md [hunk 3]
```diff
@@ -14,6 +14,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - `underline` prop for `Link Pure`
 - `Checkbox Wrapper`, `Radio Button Wrapper`, `Popover` and `Tag Dismissible` supports `theme` prop
 - `Icon` supports `x-small` and `x-large` values of `size` prop
+- `Accordion` `size="medium"` in `compact` variant
```
Comment (denyo) [inline packages/components/CHANGELOG.md:17]: Not sure what this means.

Comment (mhaegeleMHP) [reply packages/components/CHANGELOG.md]: Size medium was only available when `compact="false"`. I wrote it a bit different to clarify

File: packages/components/src/components/accordion/accordion-styles.ts (hunk 1 above)
Comment (mhaegeleMHP) [reply packages/components/src/components/accordion/accordion-styles.ts]: You want the transition on hover and when the mouse moves away from the element. If we put it into the hover media query we only have the transition on hover, not when it moves out.

Comment (denyo) [reply packages/components/src/components/accordion/accordion-styles.ts:58]: No, that's the case when you put it into the `:hover` pseudo selector. The hover media query is supposed to wrap the pseudo selector and should also wrap the transition.

File: packages/components/src/components/accordion/accordion-styles.ts [hunk 4]
```diff
@@ -22,43 +25,62 @@ export const getComponentCss = (
...
+        '&::before': {
+          content: '""',
+          position: 'absolute',
+          borderRadius: borderRadiusSmall,
+          transition: getTransition('background-color'),
+          ...(compact
+            ? {
+                top: '2px',
+                bottom: '2px',
+                left: '-4px',
```
Comment (denyo) [inline packages/components/src/components/accordion/accordion-styles.ts:63]: Identical for both cases and can therefore be moved out of the condition.

**Review by denyo — APPROVED** [review]

**Pull request merged**